tensionmap.py

```diff
@@ -129,12 +135,13 @@ def tm_update(obj, context):
     weights = [0.0] * num_vertices
 
     # calculate the new weights
-    for edge in obj.data.edges:
+    global original_edge_lengths
+    for i in range(len(obj.data.edges)):
```

Comment:
I don't see why using 
```python
for i in range(len(obj.data.edges)):
    edge = obj.data.edges[i]
```

Instead of the old

```python
for edge in obj.data.edges:
```

I don't think it is more performant (I might be wrong ?), but it certainly is worse to the eye.

EDIT: I think you needed the index for line `144`. But that won't be necessary depending on what we decide (see the rest of my comments)

Reply:
Resolved by removing the *global optimization*